### PR TITLE
[MM-41057] Wire up tutorial flow

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR wires up tutorial flow.

https://user-images.githubusercontent.com/2340127/153619566-daef7cee-18fd-43d5-80ca-3c6031314850.mp4

#### Ticket Link
  Fixes: https://mattermost.atlassian.net/browse/MM-41057

#### Checklist

- [ ] Telemetry updated
- [ ] Gated by experimental feature flag
- [ ] Unit tests updated

### CODE REVIEW

**Review by abhijit-singh — APPROVED** [review]
Looks good overall. Still not able to get the tour working properly on my local env though so approving based on the screencap.

File: webapp/src/components/backstage/playbooks/playbook.tsx [hunk 1]
```diff
@@ -316,7 +318,14 @@ const Playbook = () => {
                         </OverlayTrigger>
                     </SecondaryButtonLargerRightStyled>
                     <PrimaryButtonLarger
-                        onClick={runPlaybook}
+                        onClick={
+                            playbook.title === 'Learn how to use playbooks' ? async () => {
+                                const playbookRun = await createPlaybookRun(playbook.id, currentUserId, playbook.team_id, `${currentUser.username}'s onboarding run`, playbook.description);
+                                const pathname = `/${team.name}/channels/${playbookRun.channel_id}`;
+                                const search = '?forceRHSOpen&&openTakeATourDialog';
```
Comment (calebroseland) [inline webapp/src/components/backstage/playbooks/playbook.tsx:325]: extra `&`
```suggestion
                                const search = '?forceRHSOpen&openTakeATourDialog';
```

**Review by calebroseland — CHANGES_REQUESTED** [review]

File: webapp/src/components/rhs/rhs_run_details.tsx [hunk 2]
```diff
@@ -40,8 +41,15 @@ const RHSRunDetails = () => {
...
-        if (runDetailsStep === null && isRunDetailTour) {
+        let isRunDetailTour = false;
+        const url = new URL(window.location.href);
+        const searchParams = new URLSearchParams(url.searchParams);
+        if (searchParams.has('openTakeATourDialog')) {
+            isRunDetailTour = true;
+            searchParams.delete('openTakeATourDialog');
+            browserHistory.replace({pathname: url.pathname, search: searchParams.toString()});
+        }
+        if ((runDetailsStep === null || parseInt(runDetailsStep, 10) === FINISHED) && isRunDetailTour) {
```
[the remark this answers is not in the document]
Comment (calebroseland) [reply webapp/src/components/rhs/rhs_run_details.tsx:52]: Ahh, the initial routing over in `playbook.tsx` hits `/<team>/channels/channel_id`, and a mattermost-webapp redirect grabs it and discards the search params before `rhs_run_details` get's a chance to see them.


CleanShot 2022-02-14 at 17 00 39@2x
CleanShot 2022-02-14 at 17 00 56@2x

File: webapp/src/components/backstage/playbooks/playbook.tsx [hunk 3]
```diff
@@ -325,7 +336,7 @@ const Playbook = () => {
                             path={mdiClipboardPlayOutline}
                             size={1.25}
                         />
-                        {formatMessage({defaultMessage: 'Run'})}
+                        {playbook.title === 'Learn how to use playbooks' ? formatMessage({defaultMessage: 'Start a test run'}) : formatMessage({defaultMessage: 'Run'})}
```
Comment (crspeller) [inline webapp/src/components/backstage/playbooks/playbook.tsx:339]: Here and above maybe naming the string constant would make it clearer what is happening here?

File: webapp/src/components/rhs/rhs_run_details.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (calebroseland) [reply webapp/src/components/rhs/rhs_run_details.tsx:52]: It was only reproducible when the current channel was _not_ an onboarding run channel

**Review by calebroseland — APPROVED** [review]

**Review by crspeller — APPROVED** [review]

**Review by Phrynobatrachus — APPROVED** [review]

File: webapp/src/components/templates/template_selector.tsx [hunk 4]
```diff
@@ -56,26 +66,54 @@ const SelectorGrid = styled.div`
...
+                            reminder_timer_default_seconds: 86400,
+                            members: [{user_id: currentUserId, roles: [PlaybookRole.Member, PlaybookRole.Admin]}],
+                            team_id: teams[0].id || '',
+                            public: true,
+                        };
+
+                        const data = await savePlaybook(pb);
+                        navigateToPluginUrl(`/playbooks/${data?.id}`);
+                    };
+                }
```
Comment (calebroseland) [inline webapp/src/components/templates/template_selector.tsx:103]: @iomodo Is this going to undo https://github.com/mattermost/mattermost-plugin-playbooks/pull/1034/files#diff-ff3ba23a0c4550c04d46f67b7e68a9a5c9e08f785a6193d8238640fdd1be7804R95-R100 ?

**Pull request merged**